static/src/javascripts/projects/common/modules/ui/first-pv-consent-banner.js

```diff
@@ -104,7 +104,10 @@ const trackInteraction = (interaction: string): void => {
 };
 
 const canShow = (): Promise<boolean> =>
-    Promise.resolve([hasUnsetAdChoices(), isInEU()].every(_ => _ === true));
+    Promise.resolve(
+        [hasUnsetAdChoices(), isInEU()].every(_ => _ === true) &&
+            hasUserAcknowledgedBanner(messageCode)
+    );
```

Comment:
Should this be `!hasUserAcknowledgedBanner(messageCode)`? As in `canShow` should resolve with `true` if the user has *not* already acknowledged the banner?

It could also be written as:

`Promise.resolve([hasUnsetAdChoices(), isInEU(), !hasUserAcknowledgedBanner(messageCode)].every(_ => _ === true))`

Reply:
yeah - thanks!

do you think:
```
hasUnsetAdChoices() && isInEU() && !hasUserAcknowledgedBanner(messageCode)
```
is preferable?

Reply:
Yep I think that's preferable 👍